src/OpenLoco/Audio/Audio.cpp

```diff
@@ -1052,25 +1057,7 @@ namespace OpenLoco::Audio
 
         if (playlist.size() == 0)
         {
-            if (excludeTrack == kNoSong)
-            {
-                for (auto i = 0; i < kNumMusicTracks; i++)
-                {
-                    if (i != excludeTrack)
-                    {
-                        playlist.push_back(i);
-                    }
-                }
-            }
-            else
-            {
-                const auto& mi = kMusicInfo[excludeTrack];
-                auto currentYear = getCurrentYear();
-                if (currentYear >= mi.startYear && currentYear <= mi.endYear)
-                {
-                    playlist.push_back(excludeTrack);
-                }
-            }
+            addAllSongsToPlaylist(excludeTrack, playlist);
         }
 
         auto r = std::rand() % playlist.size();
```

Comment:
the divide-by-zero really happens on this line, but it's caused by an empty playlist. my change ensures the playlist always has something in it now